src/harness/testcases/WINNF_FT_S_PCR_testcase.py

```diff
@@ -179,22 +172,93 @@ def triggerCpasAndRetrievePpaZone(self, ppa_id, ssl_cert, ssl_key):
+    uut_ppa_zone_data = uut_fad.getZoneRecords()
+    self.assertEquals(len(uut_ppa_zone_data), 1,
+                      msg='There is no single PPA Zone record received from SAS UUT')
+
+    return uut_ppa_zone_data[0]
+
+  def assertRegConditionalsForPpaRefModel(self, registration_request,
+                                          conditional_registration_data):
+    """Asserts the REG Conditionals required for PPA creation model and raises an exception
+    if any and prepares the registration request by adding required fields.
+
+    Performs the assert to check installationParam present in registrationRequests or
+    conditional registration data and raises an exception.
+    PpaCreationModel requires the input registrationRequests to have 'installationParam'.
+    But this parameter is removed for devices where conditionals are pre-loaded.
+    Adding the 'installationParam' into registrationRequests by taking the corresponding
+    values from conditionalRegistrationData.
+
+    Args:
+      registration_request: A list of individual CBSD registration
+        requests (each of which is itself a dictionary).
+      conditional_registration_data: A list of individual CBSD registration
+        data that need to be preloaded into SAS (each of which is a dictionary).
+        the fccId and cbsdSerialNumber fields are required, other fields are optional
+        but required for ppa reference model.
+
+    Raises:
+      It will throws an exception if the installationParam object and required fields is not found
+    in conditionalRegistrationData and registrationRequests for category B and A devices
+    respectively.
 
-      # Check if the retrieved FAD that has valid PPA zone record matches the PPA ID
-      # that was generated using the admin API.
-      uut_ppa_zone_data = uut_fad.getZoneRecords([lambda x: x['id'] == ppa_id])
-      self.assertEquals(len(uut_ppa_zone_data), 1,
-                        msg='There is no single PPA Zone record matches with PPA ID '
-                            '{0} received from SAS UUT'.format(ppa_id))
-    except KeyError:
-      self.fail('There is an error while filtering PPA Zone record based '
-                'on PPA ID:{0}'.format(ppa_id))
-    else:
-      return uut_ppa_zone_data[0]
+    """
+
+    for device in registration_request:
+      if 'installationParam' not in device:
+        for conditional_params in conditional_registration_data:
+          # Check if FCC_ID+Serial_Number present in registrationRequest
+          # and conditional_params match and add the 'installationParam'.
+          self.assertIn('installationParam', conditional_params,
+                        msg='installationParm Object is not found in REG-Conditionals')
+          if not (conditional_params['fccId'] == device['fccId'] and \
+                  conditional_params['cbsdSerialNumber'] == device['cbsdSerialNumber']):
+            raise Exception('ConfigError:Wrong REG-Conditional data for device is found. '
```

Comment:
I don't think this is really what you want. Wouldn't it always raise an exception when you have more than one Cat B device?

I think you want

```
if not (...):
  break
```

so that you get to the next `conditional_param`.

Reply:
Corrected.   We have refactor the code to traverse all elements in conditionals list to   confirm whether it belongs to corresponding device rather than raising an   exception.